File: Blackjack.js
// Define card suits and ranks
const suits = ["Hearts", "Diamonds", "Clubs", "Spades"];
const ranks = ["2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King", "Ace"];

// Create a deck of cards
const deck = suits.flatMap(suit => ranks.map(rank => [suit, rank]));

// Function to shuffle the deck
function shuffleDeck(deck) {
  for (let i = deck.length - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));
    [deck[i], deck[j]] = [deck[j], deck[i]];
  }
  return deck;
}

// Function to deal a hand of cards
function dealHand(deck, numCards) {
  const hand = [];
  for (let i = 0; i < numCards; i++) {
    hand.push(deck.pop());
  }
  return hand;
}

// Function to determine the value of a hand
function handValue(hand) {
  let value = 0;
  let numAces = 0;
  for (const [, rank] of hand) {
    if (rank === "Ace") {
      numAces++;
      value += 11;
    } else if (["Jack", "Queen", "King"].includes(rank)) {
      value += 10;
    } else {
      value += parseInt(rank);
    }
  }
  while (value > 21 && numAces > 0) {
    value -= 10;
    numAces--;
  }
  return value;
}

// Function to play a round of poker
function playPoker(deck) {
  // Shuffle the deck
  shuffleDeck(deck);

  // Deal two cards to the player
  const playerHand = dealHand(deck, 2);

  // Deal two cards to the dealer
  const dealerHand = dealHand(deck, 2);

  // Print the player's hand
  console.log("Your hand:", playerHand);

  // Player's turn
  while (true) {
    // Print the player's hand value
    console.log("Your hand value:", handValue(playerHand));

    // Ask the player if they want to hit or stand
    const choice = prompt("Hit or stand? (h/s): ");

    // If the player chooses to hit
    if (choice === "h") {
      // Deal a card to the player
      playerHand.push(deck.pop());
      // Check if the player has busted
      if (handValue(playerHand) > 21) {
        console.log("You busted! Dealer wins.");
        return;
      }
    } // If the player chooses to stand
    else if (choice === "s") {
      break;
    } else {
      console.log("Invalid choice. Please enter 'h' or 's'.");
    }
  }

  // Dealer's turn
  console.log("Dealer's hand:", dealerHand);
  while (handValue(dealerHand) < 17) {
    console.log("Dealer hits.");
    dealerHand.push(deck.pop());
    console.log("Dealer's hand:", dealerHand);
    if (handValue(dealerHand) > 21) {
      console.log("Dealer busted! You win.");
      return;
    }
  }

  // Determine the winner
  const playerValue = handValue(playerHand);
  const dealerValue = handValue(dealerHand);
  if (playerValue > 21) {
    console.log("You busted! Dealer wins.");
  } else if (dealerValue > 21) {
    console.log("Dealer busted! You win.");
  } else if (playerValue > dealerValue) {
    console.log("You win!");
  } else if (dealerValue > playerValue) {
    console.log("Dealer wins.");
  } else {
    console.log("Tie!.");
  }
}

// Start the game
playPoker(deck);

File: Blackjack.py
import random

# Define card suits and ranks
suits = ["Hearts", "Diamonds", "Clubs", "Spades"]
ranks = ["2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King", "Ace"]

# Create a deck of cards
deck = [(suit, rank) for suit in suits for rank in ranks]

# Function to shuffle the deck
def shuffle_deck(deck):
    random.shuffle(deck)
    return deck

# Function to deal a hand of cards
def deal_hand(deck, num_cards):
    hand = []
    for _ in range(num_cards):
        hand.append(deck.pop())
    return hand

# Function to determine the value of a hand
def hand_value(hand):
    value = 0
    num_aces = 0
    for suit, rank in hand:
        if rank == "Ace":
            num_aces += 1
            value += 11
        elif rank in ["Jack", "Queen", "King"]:
            value += 10
        else:
            value += int(rank)
    while value > 21 and num_aces > 0:
        value -= 10
        num_aces -= 1
    return value

# Function to play a round of poker
def play_poker(deck):
    # Shuffle the deck
    deck = shuffle_deck(deck)

    # Deal two cards to the player
    player_hand = deal_hand(deck, 2)

    # Deal two cards to the dealer
    dealer_hand = deal_hand(deck, 2)

    # Print the player's hand
    print("Your hand:", player_hand)

    # Player's turn
    while True:
        # Print the player's hand value
        print("Your hand value:", hand_value(player_hand))

        # Ask the player if they want to hit or stand
        choice = input("Hit or stand? (h/s): ")

        # If the player chooses to hit
        if choice == "h":
            # Deal a card to the player
            player_hand.append(deck.pop())
            # Check if the player has busted
            if hand_value(player_hand) > 21:
                print("You busted! Dealer wins.")
                return
        # If the player chooses to stand
        elif choice == "s":
            break
        else:
            print("Invalid choice. Please enter 'h' or 's'.")

    # Dealer's turn
    print("Dealer's hand:", dealer_hand)
    while hand_value(dealer_hand) < 17:
        print("Dealer hits.")
        dealer_hand.append(deck.pop())
        print("Dealer's hand:", dealer_hand)
        if hand_value(dealer_hand) > 21:
            print("Dealer busted! You win.")
            return

    # Determine the winner
    player_value = hand_value(player_hand)
    dealer_value = hand_value(dealer_hand)
    if player_value > 21:
        print("You busted! Dealer wins.")
    elif dealer_value > 21:
        print("Dealer busted! You win.")
    elif player_value > dealer_value:
        print("You win!")
    elif dealer_value > player_value:
        print("Dealer wins.")
    else:
        print("Tie!.")

# Start the game
play_poker(deck)
